app/models/csv_export.rb

```diff
@@ -0,0 +1,62 @@
+class CsvExport < ActiveRecord::Base
+  belongs_to :user
+  serialize :filters, JSON
+  STATUS_VALUES = [nil, 'pending', 'started', 'finished', 'downloaded', 'failed', 'deleted']
+
+  before_create :set_defaults
+  before_destroy :delete_file
+
+  validates :status, inclusion: { in: STATUS_VALUES }
+  delegate :email, to: :user, allow_nil: true
+
+
+  scope :old, lambda {
+    end_date = Time.now - Rails.application.config.samson.export_job.downloaded_age
+    timeout_date = Time.now - Rails.application.config.samson.export_job.max_age
+    where("(status = 'downloaded' AND updated_at <= :end_date) OR updated_at <= :timeout_date",
+    {end_date: end_date, timeout_date: timeout_date})
+  }
+
+  def status?(state)
+    'ready' == state.to_s ? ['downloaded', 'finished'].include?(status) : status == state.to_s
+  end
+  
+  def download_name
+    "deploys_#{created_at.to_s(:number)}.csv"
+  end
+
+  def path_file
+    "#{Rails.root}/export/#{id}"
+  end
+
+  def filters
+    filter = super || {}
+    if filter['deploys.created_at'].is_a?(String)
```

Comment:
Worked out the original issue I was having, apparently the tests was failing without the check for the String it fails the test because of a state issue.  When loaded from the DB, deploys.created_at is a string, once this function runs the state of object makes this a range.

refactoring into a differently named method it now works correctly without state issues.